plugins/exporter.koplugin/target/markdown.lua

```diff
@@ -0,0 +1,157 @@
+local UIManager = require("ui/uimanager")
+local util = require("ffi/util")
+local T = util.template
+local _ = require("gettext")
+
+local formatters = {
+    none = "%s",
+    bold = "**%s**",
+    italic = "*%s*",
+    bold_italic = "**_%s_**",
+    underline_markdownit = "++%s++",
+    underline_u_tag = "<u>%s</u>",
+    strikethrough = "~~%s~~",
+}
+
+-- markdown exporter
+local MarkdownExporter = require("base"):new {
+    name = "markdown",
+    extension = "md",
+}
+
+local formatter_buttons = {
+    {_("None"), "none"},
+    {_("Bold"), "bold"},
+    {_("Bold Italic"), "bold_italic"},
+    {_("Italic"), "italic"},
+    {_("Strikethrough"), "strikethrough"},
+    {_("Underline (Markdownit style, with ++)"), "underline_markdownit"},
+    {_("Underline (with <u></u> tags)"), "underline_u_tag"},
+}
+
+function MarkdownExporter:editFormatStyle(drawer_style, label)
+    local radio_buttons = {}
+    for _idx, v in ipairs(formatter_buttons) do
+        table.insert(radio_buttons, {
+            {
+                text = v[1],
+                checked = self.settings.formatting_options[drawer_style] == v[2],
+                provider = v[2],
+            },
+        })
+    end
+    UIManager:show(require("ui/widget/radiobuttonwidget"):new{
+        title_text = T(_("Formatting style for %1"), _(label)),
+        width_factor = 0.8,
+        keep_shown_on_apply = true,
+        radio_buttons = radio_buttons,
+        callback = function(radio)
+            self.settings.formatting_options[drawer_style] = radio.provider
+        end,
+    })
+end
+
+function MarkdownExporter:populateSettings()
+    local changed = false
+    if self.settings.formatting_options == nil then
+        self.settings.formatting_options = {
+            lighten = "italic",
+            underscore = "underline_markdownit",
+            strikeout = "strikethrough",
+            invert = "bold",
+        }
+        changed = true
+    end
+    if self.settings.highlight_formatting == nil then
+        self.settings.highlight_formatting = true
+        changed = true
+    end
+    if changed then
+        self:saveSettings()
+    end
+end
+
+function MarkdownExporter:getMenuTable()
+    self:populateSettings()
+    return {
+        text = _("Markdown"),
+        checked_func = function() return self:isEnabled() end,
+        sub_item_table = {
+            {
+                text = _("Export to Markdown"),
+                checked_func = function() return self:isEnabled() end,
+                callback = function() self:toggleEnabled() end,
+            },
+            {
+                text = _("Format Highlights based on style"),
+                checked_func = function() return self.settings.highlight_formatting end,
+                callback = function() self.settings.highlight_formatting = not self.settings.highlight_formatting end,
+            },
+            {
+                text = _("Lighten"),
```

Comment:
Maybe it would be useful to show current setting of the format style in the menu item?

Reply:
I thought about that. Can I generate label from a function?

Reply:
Surely I can. Gettext is also a function, right?

Reply:
I've generated labels using functions. However, it is not being updated on settings changes. What am I missing?

Reply:
Update touchmenu instance in callback.

Reply:
Thanks. Done. :D